### ISSUE TITLE
Labels for semi-supervised learning

### ISSUE THREAD

**Issue by ilteralp** [rusty1s/pytorch_geometric#1241]
## ❓ Questions & Help

Hi all, I want to apply node classification on my custom dataset where some of the nodes are unlabelled. I think [gcn.py](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/gcn.py) example is the one that I have to adapt. 

The problem is that all the nodes of Cora dataset the example used, seems to have labels between [0-6] . Did I get something wrong ? What should I be doing to create a semi-supervised dataset ?

**Comment by rusty1s** [comment]
Although the `Planetoid` datasets comes with fully-labeled nodes, only a subset of them is used during training, see [here](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/gcn.py#L61). You can follow the same procedure by creating a `train_mask` and only saving the targets of labeled nodes in `y`.

**Comment by ilteralp** [comment]
I thought masks are used for denoting which nodes to use for training and test due to the documentation on [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html?highlight=train_mask#common-benchmark-datasets). 

How should `Data` and `Data.y` be ordered to use masks properly ? For `Data` object, should I first add train nodes, then test nodes and finally the unlabeled ones ? And for `Data.y`, should it be like train labels first and then test labels ?

**Comment by rusty1s** [comment]
They do not need to be ordered. The most easiest way would be to probably have a `train_y` vector, which holds the label for each node, and nodes without any label contain values such as `-1`.
Then, in your training loop, you can do the following:
```python
out = model(...)
train_idx = data.train_y != -1
loss(out[train_idx], data.train_y[train_idx])
```

**Comment by ilteralp** [comment]
Alright, thanks a lot.
